20-ssh-and-scp/COURSE_MATERIAL.md

```diff
@@ -709,13 +572,79 @@ You should see the `remote.txt` file in the current directory with its content:
 cat remote.txt
 ```
 
+### Add the teaching staff's public key to the virtual machine
+
+In this section, you will add the teaching staff's public key to the virtual
+machine. This will allow the teaching staff to connect to the virtual machine
+with SSH and help you if needed.
+
+The teaching staff's public key is the following:
+
+```text
+ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIF5deyMbQaRaiO4ojymkCoWBtwPyG8Y+4BbLQsb413KC heig-vd-dai-course
+```
+
+You can add the public key to the `~/.ssh/authorized_keys` file of the user on
+the virtual machine using the following command:
+
+```sh
+# Export the public key to the authorized_keys file
+export TEACHING_STAFF_SSH_PUBLIC_KEY="ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIF5deyMbQaRaiO4ojymkCoWBtwPyG8Y+4BbLQsb413KC heig-vd-dai-course"
+
+# Add the teaching staff's public key to the authorized_keys file
+echo "${TEACHING_STAFF_SSH_PUBLIC_KEY}" | ssh ubuntu@<vm public ip> "cat >> ~/.ssh/authorized_keys"
+```
+
+SSH can be used to execute a command on a remote machine. In this case, the
+`cat` command is executed on the remote machine to append the public key to the
+`~/.ssh/authorized_keys` file.
+
+As mentioned before, the public key can be shared with anyone. The private key
+must be kept secret.
+
+By executing the command above, you are allowing the teaching staff to connect
+to the virtual machine with the private key associated with the public key.
+
+> [!TIP]
+>
+> You can also add the public key to the `~/.ssh/authorized_keys` file
+> automatically with the help of the
+> [`ssh-copy-id`](https://man7.org/linux/man-pages/man1/ssh-copy-id.1.html)
+> command. You must however have the private key associated with the public key
+> to use this command. This is why you had to do it manually in this case.
+
+You can check the `~/.ssh/authorized_keys` file to verify that the public key
+has been added to the file.
+
+If you ever need to revoke the public key, you can remove it from the
+`~/.ssh/authorized_keys` file with the help of an editor such as Vi/Vim or Nano.
+
+Revoking a public key is as easy as removing it from the `authorized_keys` file.
+
+### Optional: shutdown the virtual machine
+
+If you want to stop the virtual machine to avoid consuming your free credits,
+you can stop the virtual machine with the following command:
+
+```sh
+# Stop the virtual machine
+sudo poweroff
+```
```

Comment:
I would add an important note on the fact that students MUST stop their VM to avoid consuming credits unnecessarily once the practical content has been completed.

Reply:
As the current virtual machine "only" consumes between 5 to 10 dollars per month, I considered this negligible to stop it every time it is not in use. 

Even more as the VM doesn't cost for the first 750 hours and the 100 dollars credits are only valid 12 months.

At least, we know the VM would be accessible at any time as we might need to access it for the corrections or if any help is needed.

But do you think we should insist on that even with these statements?

Reply:
I did find a phrasing that should be good now!